src/main/java/uk/gov/hmcts/ccd/data/definition/CachedCaseDefinitionRepository.java

```diff
@@ -49,12 +56,18 @@ public CachedCaseDefinitionRepository(@Qualifier(DefaultCaseDefinitionRepository
     @Override
     public CaseTypeDefinition getCaseType(final String caseTypeId) {
         CaseTypeDefinitionVersion latestVersion = this.getLatestVersion(caseTypeId);
-        return caseDefinitionRepository.getCaseType(latestVersion.getVersion(), caseTypeId);
+        return this.getCaseType(latestVersion.getVersion(), caseTypeId);
     }
 
     @Override
     public CaseTypeDefinition getCaseType(int version, String caseTypeId) {
-        return caseDefinitionRepository.getCaseType(version, caseTypeId);
+        try {
+            return caseTypes.computeIfAbsent(caseTypeId + version,
+                e -> caseDefinitionRepository.getCaseType(version, caseTypeId)).shallowClone();
```

Comment:
any particular reason for clone?

Reply:
Yes, the caller is mutating the cached object. I thought to clone the object as we don't need to worry about anyone changing the cached object in the future.